src/registrar/models/domain.py

```diff
@@ -1029,13 +1174,71 @@ def _fetch_cache(self, fetch_hosts=False, fetch_contacts=False):
                     cleaned["hosts"].append(
                         {k: v for k, v in host.items() if v is not ...}
                     )
-
             # replace the prior cache with new data
             self._cache = cleaned
 
         except RegistryError as e:
             logger.error(e)
 
+    def _get_or_create_public_contact(self, public_contact: PublicContact):
+        """Tries to find a PublicContact object in our DB.
+        If it can't, it'll create it."""
+        db_contact = PublicContact.objects.filter(
+            registry_id=public_contact.registry_id,
+            contact_type=public_contact.contact_type,
+            domain=self,
+        )
+
+        # Raise an error if we find duplicates.
+        # This should not occur...
+        if db_contact.count() > 1:
+            raise Exception(
+                f"Multiple contacts found for {public_contact.contact_type}"
+            )
+
+        if db_contact.count() == 1:
+            existing_contact = db_contact.get()
+            # Does the item we're grabbing match
+            # what we have in our DB?
+            # If not, we likely have a duplicate.
+            if (
+                existing_contact.email != public_contact.email
+                or existing_contact.registry_id != public_contact.registry_id
+            ):
```

Comment:
Not opposed to that. The only case in which this could trigger is if EppLib makes an update on their end and its not updated for us, or if somehow a user goes in and makes changes through our DB (not with our setter).